alexandria/core/models.py

```diff
@@ -1,14 +1,22 @@
 import re
+import os
 import uuid
 
 from django.core.exceptions import ImproperlyConfigured
 from django.core.validators import RegexValidator
 from django.db import models
+from django.dispatch import receiver
+from django.forms.models import inlineformset_factory
 from django.utils.translation import gettext_lazy as _
 from localized_fields.fields import LocalizedCharField, LocalizedTextField
 from rest_framework import exceptions
 
-from .storage_clients import client
+
+def file_directory_path(_file, filename):
+    # directory for File uploads to the local filesystem.
+    return "file/files/{0}/{1}{2}".format(
+        _file.document.pk, str(_file.pk)[:8], filename
```

Comment:
Django File field seems to have built-in filename collision safety, so this should be enough:
```suggestion
    return "file/files/{0}/{1}".format(
        _file.document.pk, filename
```

Reply:
It tried uploading two files with the same name in our legacy documents module, which runs with similar config. The second file had some random characters appended to its name to resolve the collision, so I think we don’t have to solve this ourselves.

Reply:
done